jetstream, invalid memory / nil pointer dereference, update a non-existent stream

## Defect

When sending an update stream message to the JetStream server and the stream does not exist. Expecting some sort of error message. Instead server crashes. 

#### Versions/OS/Environment:

Direct build `go build main.go` of source. Able to reproduce on both windows and ubuntu. Single (not clustered) server:

```
Ubuntu 20.04.1 LTS
nats-server version 2.2.0-beta.44
Go version go1.15.7 linux/amd64
nats-server -DV -p 4222 -js -sd /home/scottf/nats/jetstream-storage
```

```
Windows 10 Pro, Version 1909, Build 18363.1216
nats-server version 2.2.0-beta.42 
Go version go1.15.3 windows/amd64
nats-server -DV -p 4222 -js -sd C:\nats\jetstream-storage
```


#### Steps or code to reproduce the issue:
Using the java client (new code) sent this message:

```
PUB $JS.API.STREAM.UPDATE.bugStreamName _INBOX.gnlEkQi5jTKYwCCR0Z4SsB 138␍␊{"name":"bugStreamName","subjects":["bugSubject"],"retention":"limits","storage":"memory","discard":"old","num_replicas":1,"no_ack":false}␍␊
```

#### Expected result:
Some error response.

#### Actual result:
Server fault.

Ubuntu:

```
[45705] 2021/01/20 11:35:00.981467 [TRC] 192.168.50.43:57882 - cid:4 - "v2.9.0:java" - <<- [PUB $JS.API.STREAM.UPDATE.bugStreamName _INBOX.yZ0KBPDXfzWKCNjSGLTpHT 138]
[45705] 2021/01/20 11:35:00.981499 [TRC] 192.168.50.43:57882 - cid:4 - "v2.9.0:java" - <<- MSG_PAYLOAD: ["{\"name\":\"bugStreamName\",\"subjects\":[\"bugSubject\"],\"retention\":\"limits\",\"storage\":\"memory\",\"discard\":\"old\",\"num_replicas\":1,\"no_ack\":false}"]
[45705] 2021/01/20 11:35:00.981849 [DBG] 192.168.50.43:57882 - cid:4 - "v2.9.0:java" - Auto-unsubscribe limit of 1 reached for sid '3'
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x992f7b]

goroutine 121 [running]:
github.com/nats-io/nats-server/v2/server.(*Stream).Config(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/scottf/nats/nats-server/server/stream.go:635 +0x5b
github.com/nats-io/nats-server/v2/server.(*Stream).Update(0x0, 0xc0003ba000, 0xc0000ea000, 0xc00025a480)
	/home/scottf/nats/nats-server/server/stream.go:671 +0x14d
github.com/nats-io/nats-server/v2/server.(*Server).jsStreamUpdateRequest(0xc0000c1500, 0xc000131ce0, 0xc00019ac80, 0xc00025a480, 0x23, 0xc000218ee0, 0x11, 0xc0003b8000, 0x172, 0x31a)
	/home/scottf/nats/nats-server/server/jetstream_api.go:881 +0x707
github.com/nats-io/nats-server/v2/server.(*client).deliverMsg(0xc00019ac80, 0xc000131ce0, 0xc00025a450, 0x23, 0x30, 0xc000218ec0, 0x11, 0x20, 0xc00019c0e8, 0x47, ...)
	/home/scottf/nats/nats-server/server/client.go:3017 +0x36f
github.com/nats-io/nats-server/v2/server.(*client).processMsgResults(0xc00019ac80, 0xc00000c1e0, 0xc000211440, 0xc0003b8000, 0x174, 0x31a, 0x0, 0x0, 0x0, 0xc00025a450, ...)
	/home/scottf/nats/nats-server/server/client.go:3875 +0x62b
github.com/nats-io/nats-server/v2/server.(*client).processServiceImport(0xc00019ac80, 0xc0001bc870, 0xc0000ea000, 0xc0003b8000, 0x174, 0x31a)
	/home/scottf/nats/nats-server/server/client.go:3696 +0xa25
github.com/nats-io/nats-server/v2/server.(*Account).addServiceImportSub.func1(0xc0001b8c60, 0xc00019ac80, 0xc00025a3c0, 0x23, 0xc000218e40, 0x1d, 0xc00019aecd, 0x8c, 0xfbb)
	/home/scottf/nats/nats-server/server/accounts.go:1807 +0x5e
github.com/nats-io/nats-server/v2/server.(*client).deliverMsg(0xc00019ac80, 0xc0001b8c60, 0xc00019ae88, 0x23, 0x1000, 0xc00019aeac, 0x1d, 0xfdc, 0xc00019c0e9, 0x4d, ...)
	/home/scottf/nats/nats-server/server/client.go:3015 +0x47d
github.com/nats-io/nats-server/v2/server.(*client).processMsgResults(0xc00019ac80, 0xc0000ea000, 0xc0002113e0, 0xc00019aecd, 0x8c, 0xfbb, 0x0, 0x0, 0x0, 0xc00019ae88, ...)
	/home/scottf/nats/nats-server/server/client.go:3875 +0x62b
github.com/nats-io/nats-server/v2/server.(*client).processInboundClientMsg(0xc00019ac80, 0xc00019aecd, 0x8c, 0xfbb, 0xc00019ae50)
	/home/scottf/nats/nats-server/server/client.go:3450 +0x3be
github.com/nats-io/nats-server/v2/server.(*client).processInboundMsg(0xc00019ac80, 0xc00019aecd, 0x8c, 0xfbb)
	/home/scottf/nats/nats-server/server/client.go:3316 +0x95
github.com/nats-io/nats-server/v2/server.(*client).parse(0xc00019ac80, 0xc000384a00, 0x7, 0x200, 0x7, 0x0)
	/home/scottf/nats/nats-server/server/parser.go:468 +0x24fb
github.com/nats-io/nats-server/v2/server.(*client).readLoop(0xc00019ac80, 0x0, 0x0, 0x0)
	/home/scottf/nats/nats-server/server/client.go:1092 +0x568
github.com/nats-io/nats-server/v2/server.(*Server).createClient.func2()
	/home/scottf/nats/nats-server/server/server.go:2302 +0x45
created by github.com/nats-io/nats-server/v2/server.(*Server).startGoRoutine
	/home/scottf/nats/nats-server/server/server.go:2685 +0xc5
```

Windows:
```
[2492] 2021/01/20 08:00:32.886004 [TRC] 127.0.0.1:52198 - cid:4 - "v2.9.0:java" - <<- [PUB $JS.API.STREAM.UPDATE.bugStreamName _INBOX.B70zqeU5pelh7Ropg7dlvw 138]
[2492] 2021/01/20 08:00:32.886004 [TRC] 127.0.0.1:52198 - cid:4 - "v2.9.0:java" - <<- MSG_PAYLOAD: ["{\"name\":\"bugStreamName\",\"subjects\":[\"bugSubject\"],\"retention\":\"limits\",\"storage\":\"memory\",\"discard\":\"old\",\"num_replicas\":1,\"no_ack\":false}"]
[2492] 2021/01/20 08:00:32.887002 [DBG] 127.0.0.1:52198 - cid:4 - "v2.9.0:java" - Auto-unsubscribe limit of 1 reached for sid '3'
[2492] 2021/01/20 08:00:32.887002 [TRC] 127.0.0.1:52198 - cid:4 - "v2.9.0:java" - ->> [MSG _INBOX.B70zqeU5pelh7Ropg7dlvw 3 136]
[2492] 2021/01/20 08:00:32.887002 [TRC] 127.0.0.1:52198 - cid:4 - "v2.9.0:java" - <-> [DELSUB 3]
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0xa7ad22]

goroutine 20 [running]:
github.com/nats-io/nats-server/v2/server.(*Stream).Config(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        C:/nats/nats-server/server/stream.go:613 +0x62
github.com/nats-io/nats-server/v2/server.(*Stream).Update(0x0, 0xc0001a6780, 0xc0001dc000, 0xc0001bf050)
        C:/nats/nats-server/server/stream.go:649 +0x14d
github.com/nats-io/nats-server/v2/server.(*Server).jsStreamUpdateRequest(0xc0001b1500, 0xc0001fda20, 0xc000152000, 0xc0001bf050, 0x23, 0xc000276080, 0x11, 0xc000278000, 0x16c, 0x30e)
        C:/nats/nats-server/server/jetstream_api.go:880 +0x6e7
github.com/nats-io/nats-server/v2/server.(*client).deliverMsg(0xc000152000, 0xc0001fda20, 0xc0001bf020, 0x23, 0x30, 0xc000276060, 0x11, 0x20, 0xc000153468, 0x47, ...)
        C:/nats/nats-server/server/client.go:3017 +0x37a
github.com/nats-io/nats-server/v2/server.(*client).processMsgResults(0xc000152000, 0xc0001dc1e0, 0xc000229170, 0xc000278000, 0x16e, 0x30e, 0x0, 0x0, 0x0, 0xc0001bf020, ...)
        C:/nats/nats-server/server/client.go:3875 +0x62b
github.com/nats-io/nats-server/v2/server.(*client).processServiceImport(0xc000152000, 0xc0002267e0, 0xc0001dc000, 0xc000278000, 0x16e, 0x30e)
        C:/nats/nats-server/server/client.go:3696 +0xa25
github.com/nats-io/nats-server/v2/server.(*Account).addServiceImportSub.func1(0xc0002209a0, 0xc000152000, 0xc0001bef90, 0x23, 0xc00018bfe0, 0x1d, 0xc00015224d, 0x8c, 0xfbb)
        C:/nats/nats-server/server/accounts.go:1803 +0x65
github.com/nats-io/nats-server/v2/server.(*client).deliverMsg(0xc000152000, 0xc0002209a0, 0xc000152208, 0x23, 0x1000, 0xc00015222c, 0x1d, 0xfdc, 0xc000153469, 0x4d, ...)
        C:/nats/nats-server/server/client.go:3015 +0x482
github.com/nats-io/nats-server/v2/server.(*client).processMsgResults(0xc000152000, 0xc0001dc000, 0xc0002290e0, 0xc00015224d, 0x8c, 0xfbb, 0x0, 0x0, 0x0, 0xc000152208, ...)
        C:/nats/nats-server/server/client.go:3875 +0x62b
github.com/nats-io/nats-server/v2/server.(*client).processInboundClientMsg(0xc000152000, 0xc00015224d, 0x8c, 0xfbb, 0xc0000802c0)
        C:/nats/nats-server/server/client.go:3450 +0x3de
github.com/nats-io/nats-server/v2/server.(*client).processInboundMsg(0xc000152000, 0xc00015224d, 0x8c, 0xfbb)
        C:/nats/nats-server/server/client.go:3316 +0xa5
github.com/nats-io/nats-server/v2/server.(*client).parse(0xc000152000, 0xc00021a500, 0x47, 0x100, 0x47, 0x0)
        C:/nats/nats-server/server/parser.go:468 +0x251b
github.com/nats-io/nats-server/v2/server.(*client).readLoop(0xc000152000, 0x0, 0x0, 0x0)
        C:/nats/nats-server/server/client.go:1092 +0x568
github.com/nats-io/nats-server/v2/server.(*Server).createClient.func2()
        C:/nats/nats-server/server/server.go:2302 +0x4c
created by github.com/nats-io/nats-server/v2/server.(*Server).startGoRoutine
        C:/nats/nats-server/server/server.go:2685 +0xca

[process exited with code 2]
```

--- a/server/jetstream_api.go
+++ b/server/jetstream_api.go
@@ -876,6 +876,7 @@ func (s *Server) jsStreamUpdateRequest(sub *subscription, c *client, subject, re
 	if err != nil {
 		resp.Error = jsNotFoundError(err)
 		s.sendAPIResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
+		return
 	}
 
 	if err := mset.Update(&cfg); err != nil {

--- a/test/jetstream_test.go
+++ b/test/jetstream_test.go
@@ -7053,6 +7053,25 @@ func TestJetStreamRequestAPI(t *testing.T) {
 		t.Fatalf("Did not receive correct response: %+v", scResp.Error)
 	}
 
+	// Check that updating a non existing stream fails
+	cfg := server.StreamConfig{
+		Name:     "UNKNOWN_STREAM",
+		Storage:  server.FileStorage,
+		Subjects: []string{"foo"},
+	}
+	req, err = json.Marshal(cfg)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	resp, _ = nc.Request(fmt.Sprintf(server.JSApiStreamUpdateT, cfg.Name), req, time.Second)
+	scResp.Error, scResp.StreamInfo = nil, nil
+	if err := json.Unmarshal(resp.Data, &scResp); err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	if scResp.StreamInfo != nil || scResp.Error == nil || scResp.Error.Code != 404 {
+		t.Fatalf("Unexpected error: %+v", scResp.Error)
+	}
+
 	// Now lookup info again and see that we can see the new stream.
 	resp, err = nc.Request(server.JSApiAccountInfo, nil, time.Second)
 	if err != nil {

